src/mon/OSDMonitor.cc

```diff
@@ -4837,40 +4857,40 @@ int OSDMonitor::prepare_pool_stripe_width(const unsigned pool_type,
   return err;
 }
 
-int OSDMonitor::prepare_pool_crush_ruleset(const unsigned pool_type,
+int OSDMonitor::prepare_pool_crush_rule(const unsigned pool_type,
 					   const string &erasure_code_profile,
-					   const string &ruleset_name,
-					   int *crush_ruleset,
+					   const string &rule_name,
+					   int *crush_rule,
 					   ostream *ss)
 {
 
-  if (*crush_ruleset < 0) {
+  if (*crush_rule < 0) {
     switch (pool_type) {
     case pg_pool_t::TYPE_REPLICATED:
       {
-	if (ruleset_name == "") {
-	  //Use default ruleset
-	  *crush_ruleset = osdmap.crush->get_osd_pool_default_crush_replicated_ruleset(g_ceph_context);
-	  if (*crush_ruleset < 0) {
-	    // Errors may happen e.g. if no valid ruleset is available
-	    *ss << "No suitable CRUSH ruleset exists, check "
+	if (rule_name == "") {
+	  //Use default rule
+	  *crush_rule = osdmap.crush->get_osd_pool_default_crush_replicated_ruleset(g_ceph_context);
```

Comment:
My thinking is to leave most of the crush ruleset stuff alone until it's been long enough that we can remove it entirely.  Ceph can be restricted to a 'rule' concept and reject crush maps with rulesets containing more than one rule.